### ISSUE TITLE
sam delete needs to care about profile defined in the samconfig

### ISSUE THREAD

**Issue by organom** [aws/aws-sam-cli#3621]
### Description:
With an environment defined in the samconfig.saml, that requires a profile different from the default aws profile, running `sam delete --config-env ENV_FROM_SAMCONFIG --stack-name STACK_TO_DELETE` will fail because the profile is not taken into account.
If running with `AWS_PROFILE=XXX` before the above sam delete command, the environment is found and sam can proceed.
`sam deploy` behaves correctly in the system.

### Steps to reproduce:
On macOS Monterey 12.2  with sam installed over brew (v 1.37.0)
Use an environment from samconfig.saml that requires a profile (different from the AWS default profile)
Run `sam delete --config-env ENV_FROM_SAMCONFIG --stack-name STACK_TO_DELETE`

### Observed result:
`Error: The input stack STACK_TO_DELETE does not exist on Cloudformation in the region REGION_FROM_ENV`

### Expected result:
Same behavior as when adding AWS_PROFILE=XXXX  before running the sam delete command.

**Comment by terusaku** [comment]
Hello.
Probably, almost same issue happened to me.
When `sam delete` was executed with `--profile` and no samconfig.toml, region variable seems to be loaded implicitly without profile.

### pre-requirement:
- SAM CLI, version 1.72.0
- no `samconfig.toml`.
- executing `sam deploy/delete` command with `--profile` explicitly
- my profile is below.
```
    [profile poweruse]
    sso_start_url = https://d-###.awsapps.com/start
    sso_region = ap-northeast-1
    sso_account_id = #####
    sso_role_name = PowerUserAccess
    region = ap-northeast-1
    output = json
```

### reproduce detail:
- `sam deploy` works with `--profile` option,  but `sam delete` with `--profile` cause missing of region variable defined in ~/.aws/config.
    - Stack delete is always executing on us-east-1.
- command example: `sam deploy`
```
$ sam deploy --stack-name ${STACK} --s3-bucket ${SAM_BUCKET} --s3-prefix ${STACK} --capabilities CAPABILITY_IAM --role-arn ${DEPLOY_ROLE} --profile poweruse
```
- command example: `sam delete`
```
$ sam delete --stack-name ${STACK} --s3-bucket ${SAM_BUCKET} --s3-prefix ${STACK} --profile poweruse --debug             
```
- when deleting, debug stdout is below.
    - if added `--region` options, above `sam delete` was succesfully completed.
```
    2023-02-10 17:38:05,833 | Using config file: samconfig.toml, config environment: default
    2023-02-10 17:38:05,833 | Expand command line arguments to:
    2023-02-10 17:38:05,834 | --stack_name=___ --s3_bucket=___ --s3_prefix=___
    2023-02-10 17:38:06,021 | Config file location: /...../samconfig.toml
    2023-02-10 17:38:06,021 | Config file '/...../samconfig.toml' does not exist
            Are you sure you want to delete the stack scheduledCurNotify in the region us-east-1 ? [y/N]: N
```

**Comment by jfuss** [comment]
Patch is released in [v1.87.0](https://github.com/aws/aws-sam-cli/releases/tag/v1.87.0). Closing
